portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,10 +34,61 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private String name;
+  private ArrayList<String> comments = new ArrayList<String>();
+  private List<Task> tasks = new ArrayList<>();
+  private static final String TEXT_INPUT = "text-input";
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String name = "Bless";
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String title = (String) entity.getProperty("title");
+      System.out.println("title" + title);
+      long timestamp = (long) entity.getProperty("timestamp");
+      System.out.println("time" + timestamp);
+      Task task = new Task(id, title, timestamp);
+      tasks.add(task);
+    }
+    String json = convertToJson(tasks);
+    response.setContentType("text/html;");
+    response.getWriter().println(json);
+  }
+
+  private String convertToJson(List<Task> tasks) {
+    Gson gson = new Gson();
+    String json = gson.toJson(tasks);
+    return json;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, TEXT_INPUT, "");
+    System.out.println("text" + text);
+    String title = request.getParameter(TEXT_INPUT);
+    long timestamp = System.currentTimeMillis();
+
+    Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("title", text);
+    taskEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
+    // Respond with the result.
     response.setContentType("text/html;");
-    response.getWriter().println(name);
+    response.sendRedirect("index.html#comments");
```

Comment:
try to remove the `#comments` from the redirect URL, I wonder if this causes the page not to reload, and instead just jump to the anchor